src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingTemplateToken.cs

```diff
@@ -1,32 +1,271 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Globalization;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.CSharp.RuntimeBinder;
+using Newtonsoft.Json.Linq;
+
 namespace Microsoft.Azure.WebJobs.Host.Bindings.Path
 {
     /// <summary>
     /// Represents a semantic token as generated by <see cref="BindingTemplateParser"/> during parsing of parameterized 
     /// binding template string. Then generated token may be consumed to create instance of 
     /// <see cref="BindingTemplate"/> or <see cref="BindingTemplateSource"/>.
+    /// The token may be a literal value or an expression. 
     /// </summary>
-    internal struct BindingTemplateToken
+    internal abstract class BindingTemplateToken
     {
-        private readonly string _value;
-        private readonly bool _isParameter;
+        // If this is expression, get the prameter name 
```

Comment:
for 'x.y,z', it returns 'x'.  It's providing the top-level key that we use for static validation.   Since the expression can be dynamic (request.query.name1), we can only validate the first term. 

---
In reply to: [114848605](https://github.com/Azure/azure-webjobs-sdk/pull/1117#discussion_r114848605) [](ancestors = 114848605)